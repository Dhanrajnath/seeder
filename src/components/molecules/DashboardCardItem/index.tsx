import { Box, styled } from "@mui/material";
import theme from "../../../utils/theme";
import IconFromSvg from "../../atoms/IconFromSvg";
import TypographyWithInfoIcon from "../TypographyWithInfoIcon";
import Typography from "../../atoms/Typography";

export interface IDashboardCardItemProps {
    icon: string;
    infoText: string;
    title: string
};

const StyledOuterBox = styled(Box)({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    width: theme.spacing(51),
    height: theme.spacing(42),
    gap: theme.spacing(2),
    background: theme.palette.background.paper
});

const StyledInnerBox = styled(Box)({
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    width: theme.spacing(20),
    height: theme.spacing(20),
    border: theme.spacing(1),
    background: theme.palette.common.black,
    borderRadius: theme.spacing(2)
});

const DashboardCardItem = ({ icon, infoText, title }: IDashboardCardItemProps) => {
    return (
        <StyledOuterBox>
            <StyledInnerBox>
                <IconFromSvg path={icon} alt={title} />
            </StyledInnerBox>
            <TypographyWithInfoIcon title={infoText} isHeading={false} />
            <Typography variant="h1" children={title} color={theme.palette.text.primary} />
        </StyledOuterBox>
    );
};

export default DashboardCardItem;