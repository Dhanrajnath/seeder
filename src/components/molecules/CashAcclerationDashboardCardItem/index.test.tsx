import { render } from "@testing-library/react";
import CalendarSvg from "../../../../public/assets/calendar.svg";
import DashboardCardItem from ".";

describe('DashboardCardItem test', () => {
    it('renders card item', () => {
        const { container, getByText, getByAltText } = render(<DashboardCardItem icon={CalendarSvg} infoText="Term cap" title="12 months" />);

        expect(getByAltText('12 months')).toBeInTheDocument();
        expect(getByText('Term cap')).toBeInTheDocument();
        expect(getByText('12 months')).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });
});