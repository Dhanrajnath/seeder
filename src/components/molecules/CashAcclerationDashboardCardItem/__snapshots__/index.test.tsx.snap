// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardCardItem test renders card item 1`] = `
<div
  class="MuiBox-root css-11ctgz"
>
  <div
    class="MuiBox-root css-uldm2x"
  >
    <img
      alt="12 months"
      height="inherit"
      src="calendar.svg"
      width="inherit"
    />
  </div>
  <div
    class="MuiBox-root css-7dey8k"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1jvmuot-MuiTypography-root"
    >
      Term cap
    </p>
    <span
      aria-label="Term cap"
      class=""
      data-mui-internal-clone-element="true"
      style="padding-top: 8px;"
    >
      <img
        alt="Term cap"
        height="inherit"
        src="info.svg"
        width="inherit"
      />
    </span>
  </div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1q5a4v6-MuiTypography-root"
  >
    12 months
  </h1>
</div>
`;
